### ISSUE TITLE
PH4 follow-up: Variational field recovery fails hard L4 stress/traction gates

### ISSUE THREAD

**Issue by Teased-oChroid-orrA** [Teased-oChroid-orrA/NetworkNeuralNetworkStressSolver#64]
Parent: #63 (Phase 4)

## Problem
Corrected Variational training minimizes atomic measure-aware Pi = U - W_ext and passes the L4 energy-health gate, but published displacement gradients/stress do not satisfy hard no-hole thresholds.

Latest controlled evidence:
- normalized Pi approximately -1.02
- energy-balance error 1.6e-3 to 4.4e-3 (health pass)
- load-transfer ratio 0.8781 (required 0.99–1.01)
- sigma_xx relative error 0.1214 (required <0.01)
- traction RMS/reference 0.1106 (required <0.01)
- 3,000 steps, larger width/depth, FD step 1e-2, 1,024 boundary points, and fixed Pi weight did not pass hard L4

## Scope
Trace and fix field-recovery accuracy without changing Pi, thresholds, or adding benchmark correction factors. Preserve neural/PINN/DEM representation. Evaluate a general linear/residual displacement path only if mathematically justified and compatible with all formulations.

## Acceptance
- Mathematical derivation and source audit recorded.
- Focused unit/regression tests pass.
- Corrected Variational no-hole runtime artifact passes all hard L4 metrics.
- Independent field validation passes.
- Existing Hybrid/Kirsch/pin-lug regressions pass.
- Downstream app-egui check/test/release build pass if public interfaces change.
- Evidence linked from parent Phase 4 manifest.

**Comment by Teased-oChroid-orrA** [comment]
Milestone: corrected Variational Pi now converges energetically. L4 health passes (energy-balance error 1.6e-3 to 4.4e-3; normalized Pi about -1.02). Hard field/traction gates still fail: best load transfer 0.8781, sigma_xx error 0.1214, traction RMS 0.1106. Controls completed: 3,000 steps, fd_h=1e-2, fixed Pi weight, 1,024 boundary points, and 32x3 MLP. No threshold or physics relaxation used. #64 remains open for field-recovery fix.

**Comment by Teased-oChroid-orrA** [comment]
Milestone: longer/capacity control attempted. Existing 3,000-step 16x2 run completed with Pi=-1.04625 but load transfer worsened to 0.6053. New 32x3/3,000 run was terminated at step 300 due excessive runtime after matching the 1,000-step trajectory; incomplete evidence only. No acceptance status changed. Single-solver process cap preserved.

**Comment by Teased-oChroid-orrA** [comment]
Further audit completed: Hooke plane-stress mapping, engineering shear convention, normalized FD coordinate scaling, boundary normals, and geometric quadrature all match mathematical definitions. No source defect found. A benchmark-specific affine correction is prohibited by #63, so no such workaround added. Remaining defect is finite neural field approximation under weak Pi; issue stays open pending a general representation/optimizer fix.

**Comment by Teased-oChroid-orrA** [comment]
Architecture implementation update:

Implemented general coordinate-residual displacement path in `ElasticityNet`.

- Zero-initialized, trainable raw normalized `[x, y, z] -> [u, v]` projection.
- Production `fwd`/`fwd_masked` keep raw coordinates alongside optional Fourier MLP input.
- Residual affects displacement only; direct stress remains MLP-produced.
- Existing embedded-input `forward` API remains unchanged.
- Skip weight/bias are registered for gradients/optimizers; width/depth/pruning preserve branch.
- New checkpoints persist branch; legacy records load with zero residual.

Proof:
- `cargo test -p pinn-solver --features ndarray-backend checkpoint::tests -- --test-threads=1` — 3 passed.
- `cargo check -p pinn-app --features ndarray-backend` — passed.

Documented in `docs/PHASE_4_IMPLEMENTATION_MANIFEST.md`. WGPU network-unit execution is host-blocked: no Metal adapter. No L4 claim: runtime ladder, independent field validation, and downstream topology gates remain open.

**Comment by Teased-oChroid-orrA** [comment]
Comprehensive progress / failure update

Implemented:

- `ElasticityNet` has a zero-initialized trainable raw-coordinate `[x_norm, y_norm, z] -> [u, v]` residual.
- Production `fwd` and `fwd_masked` retain raw coordinates alongside optional Fourier MLP input. Residual changes displacement only; direct stress remains MLP-only.
- Embedded-input `forward` remains compatible.
- Skip weight/bias participate in optimizer gradients. Width/depth growth and pruning preserve branch.
- New checkpoints persist branch; legacy records load with zero skip.

Code proof:

- `cargo test -p pinn-solver --features ndarray-backend checkpoint::tests -- --test-threads=1`: 3 passed.
- Focused interior sampling safety tests: 2 passed.
- `cargo check -p pinn-app --features ndarray-backend`: passed.
- WGPU-only network tests cannot run on host: no Metal adapter.

Controlled runtime H:

- Fixed seed / Variational `Pi=U-W_ext` / `fd_h=1e-2` / 512 interior / 256 boundary / AMR off / 1,000 steps.
- Health passes: `energy_balance_error=2.7854e-4`; hard L4 fails all five metrics.
- `sigma_xx_relative_error=0.1525`; `sigma_yy_over_ref=0.0410`; `sigma_xy_over_ref=0.0239`; `traction_rms_over_ref=0.1231`; `load_transfer_ratio=0.8049`.
- Earlier F MLP control: `sigma_xx_relative_error=0.1196`, `load_transfer_ratio=0.8838`; H is not winning and is not extended.

New diagnosis / experiment:

- H reaches normalized `Pi=-1.028`, below exact affine continuum `-1`, while independent field metrics worsen. Evidence points to sampled-functional underintegration / neural between-sample exploitation, not valid continuum solution.
- Replaced random base interior samples with geometry-generic deterministic stratification plus existing rejection fallback for holes. Focused hole-safety tests pass.
- Stratified rerun has emitted startup only; no `step 0` or final ledger. It is not evidence.

Open blockers:

1. No corrected Variational L4 pass.
2. No independent field-validation pass.
3. No AMR, single-hole/Kt, non-square, or multiple-hole acceptance runs; all gated by L4.
4. Do not close #64 or promote #63 operational status.

Artifacts: `Debug_run/phase4/H_coordinate_residual/`. Local record: `docs/PHASE_4_IMPLEMENTATION_MANIFEST.md`.

**Comment by Teased-oChroid-orrA** [comment]
Resolved. Root cause traced and fixed in db334fd (main).

**Root cause**: `UserSamplingStrategy::sample_interior`/`sample_boundary` (`crates/pinn-solver/src/user_problem.rs`) returned the byte-identical point cloud on every training step for any no-hole geometry — a deterministic stratified grid with no per-call variation, and an RNG rejection fallback reseeded from a fixed constant every call that was never even reached with zero holes to reject. `user_runner.rs`'s training loop dutifully called these fresh every step, but for the no-hole case the "resample every step" contract was a silent no-op. `InteriorEnergyTerm`/`PhysicalPotentialEnergyTerm`'s `U` and `ExternalWorkTerm`'s `W_ext` are both plain `mean(f(x_i))` Monte-Carlo estimators, unbiased only if the `x_i` genuinely vary across the optimization trajectory — with a static node set, the network could (and did) sculpt energy density artificially low at exactly those frozen nodes while the field diverged between them. This exactly explains the manifest's own prior diagnosis: sampled `Pi` settling below the true continuum affine minimum of `-1` (`-1.02` to `-1.05`), which is impossible for the real convex functional but trivial for a biased fixed-sample estimate of it.

**Fix**: `sample_interior`/`sample_boundary` now draw genuinely different points on every call via jittered stratified sampling, seeded from a per-instance atomic call counter. Same stratum/coverage structure and `ds` quadrature-weight validity; fully reproducible run-to-run. No changes to `Pi`, SAW-BRDR weighting, any threshold, or `pinn_core::sampling`/`PinLugSamplingStrategy` (Kirsch's frozen path and pin-lug's separately-tuned path — out of scope, untouched).

**Verification** (all in db334fd):
- `examples/problems/variational_no_hole_plate.toml` (resolution raised to 4096/2048/3000 to close the last bit of estimator variance): normalized `Pi=-1.000017`, all five P2-14 hard thresholds PASS — `sigma_xx_relative_error=0.0010`, `sigma_yy_over_ref=0.0007`, `sigma_xy_over_ref=0.0003`, `traction_rms_over_ref=0.0011`, `load_transfer_ratio=0.9991`.
- `validate_no_hole_fields` (previously test-only) wired into the headless no-hole output — independent field validation (separate 96x96 grid, not training points) also PASSES.
- New durable `#[ignore]`d regression test `user_problem::tests::issue_64_resampling_fix_passes_l4_and_independent_field_validation` reproduces this via a real training run — passing.
- Full workspace suite: 457 passed, 0 failed, 27 ignored (Hybrid/Kirsch/pin-lug regressions all green).
- Two unrelated pre-existing stale test literals in `network.rs` also fixed (confirmed via `git stash` to predate this session's work).
- No public/serialized interface changed — no downstream app-egui check required.

Closing. Follow-up work (AMR, hole-boundary reconciliation, real L5, non-square, multi-hole, checkpoint persistence, final operational status) is tracked as new sub-issues under #63.
